Jenkinsfile

```diff
@@ -60,12 +60,37 @@ pipeline {
                 not { branch 'develop' }
                 not { branch 'master' }
             }
-            steps { 
+            steps {
                 script {
                     utils.killOldBuilds()
                 }
             }
         }
+        stage("Make-doxygen") {
+            agent any
+            when { branch 'PR-795'}
+            steps {
+                    retry(3) { checkout scm }
+                    withCredentials([usernamePassword(credentialsId: 'a630aebc-6861-4e69-b497-fd7f496ec46b',
+                        usernameVariable: 'GIT_USERNAME', passwordVariable: 'GIT_PASSWORD')]) {
+                        sh """#!/bin/bash
+                            set -x
+                            make doxygen
+                            git config --global user.email "mc.stanislaw@gmail.com"
+                            git config --global user.name "Stan Jenkins"
+                            git checkout --detach
+                            git branch -D gh-pages
+                            git push https://${GIT_USERNAME}:${GIT_PASSWORD}@github.com/stan-dev/math.git :gh-pages
+                            git checkout --orphan gh-pages
+                            git rm --cached -r stan test lib make
```

Comment:
Do you still need to do this `git rm` step if you're pushing just the subtree?

Reply:
Nope! I'll remove this tonight